### PR TITLE
JIT `PolyhedralPrism` even more, and remove reflected lists

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by CoronelBuendia** [comment]
I know you're not supposed to use tests to time stuff... but for me the below has gone from 38.6 seconds to 5.3 seconds.

```bash
pytest tests/magnetostatics/test_polyhedral_prism.py
```

**Comment by CoronelBuendia** [comment]
The codecov patch fails mostly because of additional nb.jit removed things from the coverage

File: bluemira/magnetostatics/polyhedral_prism.py [hunk 1]
```diff
@@ -336,8 +361,8 @@ def field_fabbri(
     """
     field = np.zeros(3)
     for i, normal in enumerate(face_normals):
```
Comment (je-cook) [inline bluemira/magnetostatics/polyhedral_prism.py:363]: if I do this I get rid of that numba performance warning. I dunno if there is a better way...
```suggestion
    for i, _normal in enumerate(face_normals):
        normal = np.ascontiguousarray(_normal)
```

**Review by je-cook — APPROVED** [review]
Other than that one numba warning comment, looks good to me

**Pull request merged**